src/commands/pickFuncProcess.ts

```diff
@@ -7,61 +7,112 @@
 
-export async function pickFuncProcess(outputChannel: vscode.OutputChannel): Promise<string | undefined> {
-    outputChannel.show();
-    outputChannel.appendLine(localize('searchingFuncHost', 'Searching for Azure Functions host process...'));
-    let funcProcess: string | undefined = await getFuncPid();
+export async function pickFuncProcess(): Promise<string | undefined> {
+    let funcHostPid: string | undefined = await getFuncHostPid();
+    if (funcHostPid !== undefined) {
+        // Stop the functions host to prevent build errors like "Cannot access the file '...' because it is being used by another process."
+        await killProcess(funcHostPid);
+    }
 
-    if (funcProcess === undefined) {
-        // If the functions host isn't running, start the task for the user
-        await vscode.commands.executeCommand('workbench.action.tasks.runTask', funcHostTaskId);
+    // Start (or restart) functions host (which will also trigger a build)
+    await vscode.commands.executeCommand('workbench.action.tasks.runTask', funcHostTaskId);
 
-        const timeoutSeconds: number = 60;
-        const maxTime: number = Date.now() + timeoutSeconds * 1000;
-        while (Date.now() < maxTime) {
-            // Wait one second between each attempt
-            await new Promise((resolve: () => void): void => { setTimeout(resolve, 1000); });
+    const timeoutSeconds: number = 60;
+    const maxTime: number = Date.now() + timeoutSeconds * 1000;
+    while (Date.now() < maxTime) {
+        // Wait one second between each attempt
+        await new Promise((resolve: () => void): void => { setTimeout(resolve, 1000); });
 
-            funcProcess = await getFuncPid();
-            if (funcProcess !== undefined) {
-                return funcProcess;
-            }
+        funcHostPid = await getFuncHostPid();
+        if (funcHostPid !== undefined) {
+            return funcHostPid;
         }
-
-        throw new Error(localize('failedToFindFuncHost', 'Failed to detect running Functions host within "{0}" seconds.', timeoutSeconds));
-    } else {
-        return funcProcess;
     }
+
+    throw new Error(localize('failedToFindFuncHost', 'Failed to detect running Functions host within "{0}" seconds.', timeoutSeconds));
 }
 
-async function getFuncPid(): Promise<string | undefined> {
-    const processList: IProcess[] = await new Promise((resolve: (processList: IProcess[]) => void, reject: (e: Error) => void): void => {
-        //tslint:disable-next-line:no-unsafe-any
-        ps.lookup(
-            {
-                command: '.*dotnet.*',
-                arguments: '.*Azure\.Functions.*host.*start'
-            },
-            (error: Error | undefined, result: IProcess[]): void => {
-                if (error) {
-                    reject(error);
-                } else {
-                    resolve(result);
-                }
-            });
-    });
+function isWindows(): boolean {
+    return /^win/.test(process.platform);
+}
 
```

Comment:
How about adding some CTI test cases?  Esp. cause of the win/Linux split?

Reply:
Done